runtime/j9vm/javanextvmi.cpp

```diff
@@ -272,8 +273,24 @@ exitVThreadTransitionCritical(J9VMThread *currentThread, j9object_t vthread)
 	J9OBJECT_I64_STORE(currentThread, vthread, currentThread->javaVM->virtualThreadInspectorCountOffset, 0);
 }
 
-JNIEXPORT void JNICALL
-JVM_VirtualThreadMountBegin(JNIEnv *env, jobject thread, jboolean firstMount)
+#if JAVA_SPEC_VERSION >= 20
+static void
+virtualThreadHideFrames(J9VMThread *currentThread, j9object_t vthread, BOOLEAN hide) {
+	Assert_SC_true(IS_JAVA_LANG_VIRTUALTHREAD(currentThread, currentThread->threadObject));
+
+	if (hide) {
+		Assert_SC_true(J9_ARE_NO_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES));
+		currentThread->privateFlags |= J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
+	} else {
+		Assert_SC_true(J9_ARE_ALL_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES));
+		currentThread->privateFlags &= ~(UDATA)J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
+	}
+}
```

Comment:
Keeping in mind of the possible FastJNI code for this, I would like to see this being added to VM_VMHelpers and called by `JVM_VirtualThreadHideFrames` directly. no reason to create this helper which cannot be shared.
Note the assert will have to stay inside the JVM* function since VMHelpers cannot assert